src/core/component/q-component.public.ts

```diff
@@ -91,7 +91,8 @@ export function withScopedStyles(styles: TypeOrQRL<string>): void {
   _withStyles(styles, true);
 }
 
-type OnMountFn<PROPS> = (props: PROPS) => ReturnType<typeof onRender>;
+type MightPromise<T> = Promise<T> | T;
```

Comment:
```suggestion
type PossiblyPromise<T> = Promise<T> | T;
```
? The idea is that `MightPromise` sounds to me like a verb, while `PossiblyPromise` sounds to me a bit more like a noun.